R/tokens.R

```diff
@@ -307,36 +307,27 @@ tokens.corpus <- function(x, ..., include_docvars = TRUE) {
 #' object is a \link{tokens} object, and functions to combine \link{tokens}
 #' objects.
 #' @param x object to be coerced or checked
+#' @param concatenator character between multi-word expressions, default is _ (underscore) character
 #' @return \code{as.tokens} returns a quanteda \link{tokens} object
 #' @export
 #' @rdname as.tokens
-as.tokens <- function(x) {
+as.tokens <- function(x, concatenator = '_') {
     UseMethod("as.tokens")
 }
 
 #' @rdname as.tokens
 #' @export
-as.tokens.list <- function(x) {
+as.tokens.list <- function(x, concatenator = '_') {
     result <- tokens_hash(x)
     attr(result, "what") <- "word"
     attr(result, "ngrams") <- 1L
-    attr(result, "concatenator") <- "_"
+    attr(result, "concatenator") <- concatenator
     attr(result, 'padding') <- FALSE
     class(result)[2] <- "tokenizedTexts"
     return(result)
 }
 
 #' @export
-#' @method as.tokens collocations
```

Comment:
`textstat_collocations` no longer keep the original IDs, so there is no point to return tokens-like object from the C++ side (and there is not object to coerce). If you want to keep this function, we just tokenize characters. I will do this quickly.